packages/datetime/src/common/dateUtils.ts

```diff
@@ -110,6 +110,27 @@ export function isMonthInRange(date: Date, dateRange: DateRange) {
     return start <= day && day <= end;
 }
 
+export function isTimeInRange(date: Date, minDate: Date, maxDate: Date): boolean {
+    const time = getDateOnlyWithTime(date).getTime();
+    const minTime = getDateOnlyWithTime(minDate).getTime();
+    const maxTime = getDateOnlyWithTime(maxDate).getTime();
+
+    const isTimeGreaterThanMinTime = time >= minTime;
+    const isTimeSmallerThanMaxTime = time <= maxTime;
+
+    if (maxTime <= minTime) {
```

Comment:
Well, to be honest, I feel different about this. If `isTimeInRange` is supposed to tell me if given time is between another two times, and we consider `19:00 - 02:00` as a valid time range, then function works as expected. But I think I get your point - `maxTime > mintTime` from the logical point of view, should never happen, and that indeed is exceptional. That's why I won't argue, returning error is also fine for me :)

So, to clamp implementation details - I should catch error from `isTimeInRange` in the component, then if `allowOverlappingTimes` prop is set to `true`, set a new time, otherwise don't set a new time. Do I get it right?

Ps. Overlapping times - that's how this case should be named, not some kind of "past midnight" thing that I used to call it in tests :P I'll also fix that.